bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/IndexInMemPageMgr.java

```diff
@@ -318,7 +308,11 @@ public void onSetInUse(LedgerEntryPage lep) {
 
         @Override
         public void onResetInUse(LedgerEntryPage lep) {
-            addToCleanPagesList(lep);
+            if (!lep.isDeleted()) {
```

Comment:
Is this a bug fix?

Reply:
Not really, the isDeleted() concept came in with this patch.  It's there to deal with the inevitability of the consistency checker holding an LEP reference as the corresponding ledger is deleted.